### ISSUE TITLE
[Due for payment 2025-03-10] [$250] Workspace - Unable to scroll the page &skeleton pattern differs

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53917]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** V9. 0.74-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch app in both mweb and android
2. Go to workspace settings - categories
3. Select all category and delete it
4. Note the skeleton pattern above and below image
5. Try to scroll the page

## Expected Result:
Skeleton pattern above& below must be same and scrolling of page must also be consistent in mweb and android.

## Actual Result:
Skeleton pattern above& below is different in mweb and android. Then user can scroll the page in mweb but cannot scroll in android.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/40fe5edc-a652-4997-a1a8-b6934aa6ad8c

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869126524576530268</li>
        <li>Upwork Job ID: 1869126524576530268</li>
        <li>Last Price Increase: 2025-02-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @CortneyOfstad</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconcistency scroll behavior between native and mWeb.

### What is the root cause of that problem?
The empty view component is wrapped with a ScrollView.
https://github.com/Expensify/App/blob/890806d8cca43d1c9ae7602ccb31e4a6bc5dff4e/src/components/EmptyStateComponent/index.tsx#L85-L95

It was added for [this](https://github.com/Expensify/App/pull/44137#issuecomment-2205336599). In our case here, the view becomes scrollable when the skeleton items amount is bigger than the container can hold because we use `ceil`, so it's overflown.

https://github.com/Expensify/App/blob/890806d8cca43d1c9ae7602ccb31e4a6bc5dff4e/src/components/Skeletons/ItemListSkeletonView.tsx#L53

It behaves differently between native and mWeb because the mWeb screen height is smaller, so the skeleton amount also differs.

### What changes do you think we should make in order to solve the problem?

We have 2 options. First, we can add a new props to `ItemListSkeletonView` so we use `Math.floor` instead of `Math.ceil` so the skeleton items won't ever overflown the container.

The 2nd one is to set the skeleton background view overflow to hidden.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N/A

**Comment by CortneyOfstad** [comment]
Still working on testing this

**Comment by CortneyOfstad** [comment]
Hey @eVoloshchak! I was able to recreate, and it looks like we have a proposal [here](https://github.com/Expensify/App/issues/53917#issuecomment-2535833760) for review when you have a chance. Thanks!

**Comment by CortneyOfstad** [comment]
@eVoloshchak we have a proposal [here](https://github.com/Expensify/App/issues/53917#issuecomment-2535833760) — any thoughts?

**Comment by CortneyOfstad** [comment]
@eVoloshchak bump on the proposal review. Please complete that before EOD — thank you!

**Comment by eVoloshchak** [comment]
> we can add a new props to ItemListSkeletonView so we use Math.floor instead of Math.ceil so the skeleton items won't ever overflown the container.

@bernhardoj, all of the items already fit on the screen for me (that would heavily depend on the device I suppose)

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/281d6bcd-4c67-43e2-ae76-0695f5a549f5" />

But there is a padding below the list itself which makes it overflow. Could we remove this padding?

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/25eb10cf-878d-44fd-8d15-8536d985dec1" />

**Comment by bernhardoj** [comment]
> (that would heavily depend on the device I suppose)

Yes, you can try resizing your browser to repro it, or test it on mWeb and Android as suggested on the OP.

> But there is a padding below the list itself which makes it overflow. Could we remove this padding?

It's a skeleton item but with 0 opacity.

<img width="1351" alt="Image" src="https://github.com/user-attachments/assets/c78b9ba9-e0e4-4f1f-9ecc-a8c239ad5144" />

https://github.com/Expensify/App/blob/7b8bb554e0c7aa23c9e01682d8b63694203b8455/src/components/Skeletons/ItemListSkeletonView.tsx#L64

It's probably intentional, not sure.

**Comment by CortneyOfstad** [comment]
@eVoloshchak any follow-up on the comment [here](https://github.com/Expensify/App/issues/53917#issuecomment-2589275438)? Thanks!

**Comment by CortneyOfstad** [comment]
@eVoloshchak bump ^^^ thanks!

**Comment by CortneyOfstad** [comment]
@eVoloshchak please provide an update on the comment [here](https://github.com/Expensify/App/issues/53917#issuecomment-2589275438) by EOD.

**Comment by CortneyOfstad** [comment]
Okay, reassigning the C+ as we need movement on this.

**Comment by CortneyOfstad** [comment]
Hey @Ollyws! We have some proposals and commentary listed above. Can you take a look when you have a chance — thank you!

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Unable to scroll the page &skeleton pattern differs

### What is the root cause of that problem?

The skeleton loader component was calculating the number of items using Math.ceil(), which sometimes added an extra item that overflowed the container, causing unintended scrolling.

https://github.com/Expensify/App/blob/4c9a883227273fbe89923ed4d2236b370e673344/src/components/Skeletons/ItemListSkeletonView.tsx#L53

Problem: Math.ceil() allows one extra skeleton to be added, making the container taller than intended.
•	Impact:
•	On mWeb (smaller screens) → more overflow → unnecessary scrolling.
•	On native (larger screens) → scrolling may not appear at all.

### What changes do you think we should make in order to solve the problem?

To ensure skeleton items never exceed the available height, some thing like this:

1. Stores container height dynamically (setContainerHeight) to adjust skeleton count correctly.

```diff
  const [containerHeight, setContainerHeight] = useState(0);
```

https://github.com/Expensify/App/blob/4c9a883227273fbe89923ed4d2236b370e673344/src/components/Skeletons/ItemListSkeletonView.tsx#L52

update to:

```diff
    const availableHeight = event.nativeEvent.layout.height; // Get actual container height
    setContainerHeight(availableHeight); // Store container height for consistency
```

2 Uses Math.floor() instead of Math.ceil() to ensure no extra item is added beyond available height.

https://github.com/Expensify/App/blob/4c9a883227273fbe89923ed4d2236b370e673344/src/components/Skeletons/ItemListSkeletonView.tsx#L53

update to:

```diff
    const newNumItems = Math.floor(availableHeight / totalItemHeight); 
```
3

https://github.com/Expensify/App/blob/4c9a883227273fbe89923ed4d2236b370e673344/src/components/Skeletons/ItemListSkeletonView.tsx#L83

Update to:

```diff
    <View
        style={[
            themeStyles.flex1,
            {height: containerHeight}, // Prevent scrolling
        ]}
        onLayout={handleLayout}
    >
```

## [Test branch](https://github.com/Expensify/App/compare/main...huult:App:53917-test?expand=1)

https://github.com/user-attachments/assets/53073183-680e-488d-bdf1-c23a0b7f88a8

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


None

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by dannymcclain** [comment]
For this one I would probably opt to use our "basic" skeleton row, which looks like this:

![Image](https://github.com/user-attachments/assets/5ee20042-0f82-4974-8e19-ab9ca9634e97)

But I'll let @Expensify/design weigh in as well!

**Comment by shawnborton** [comment]
That works for me! 

As for the `floor` vs `ciel` thing - whatever gives us a clean gradient from top to bottom works for me. Maybe we can compare a side-by-side?

**Comment by dannymcclain** [comment]
>Setting the the skeleton background view overflow to hidden gets rid of the unnecessary scrolling on mWeb

And sorry, just to clarify, we agree that this page should **not** scroll, right? Was just getting a little confused by the title of the issue.

**Comment by shawnborton** [comment]
Agree. I think we'd only want it to scroll if the viewport was shorter than the height of the empty state content.

**Comment by chiragsalian** [comment]
Proposal LGTM, feel free to create the PR @bernhardoj.

**Comment by bernhardoj** [comment]
@Ollyws to clarify, which solution should I use? The overflow hidden?

**Comment by bernhardoj** [comment]
PR is ready

cc: @Ollyws 

@dannymcclain for your comment [here](https://github.com/Expensify/App/issues/53917#issuecomment-2665910292), are you suggesting to create a new skeleton, because we don't have that kind of skeleton yet?

**Comment by dannymcclain** [comment]
Oh you're right. For some reason I thought we already had that one in the product. I think it's fine to use the one that's there, but it would be a little bit better to have that basic skeleton for cases like this (and tags, taxes, etc—basically any table rows that don't have an avatar piece to them).

Let's see what @Expensify/design thinks—is it worth it to add this type of skeleton that we have in Figma but not in the product?

**Comment by shawnborton** [comment]
It would definitely be nice but we could probably do it as a follow up so that we focus on just fixing this immediate bug first?

**Comment by dannymcclain** [comment]
Sounds good!

**Comment by CortneyOfstad** [comment]
@Ollyws please complete the checklist as soon as you can so there is no delay in payment before the 10th. Thank you!

**Comment by CortneyOfstad** [comment]
## Payment Summary

- @Ollyws — to be paid $250 via NewDot
- @bernhardoj — to be paid $250 via NewDot

## Regression Test

https://github.com/Expensify/Expensify/issues/478855

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
